### ISSUE TITLE
async/await pool doesn't get closed automatically?

### ISSUE THREAD

**Issue by giridharangm** [brianc/node-postgres#1669]
I'm a bit confused about the usage of `pool.query()` here. I've set up a `query` export just like this:  

![image](https://user-images.githubusercontent.com/2258588/40908261-c1e1fd64-67d5-11e8-91ba-d237ab6e1f54.png)  

And I'm using this `query` method throughout the app. Functionally, this works great!. But, started showing some warnings once I integrated `jest`.  Like:  

![image](https://user-images.githubusercontent.com/2258588/40908319-e937e2d4-67d5-11e8-9900-9b849f9f04f9.png)  

On my debugging, I found out it's the `pool` instance that's waiting. Is that intentional?  

What am I missing?

**Comment by sehrope** [comment]
Yes that's intentional. The pool keeps some timers open while there are active connections in order to reap the connections when they expire. The default life for an idle connection is 30-seconds. After that it would be closed by the pool.

If you want to cleanly exit early call `pool.end()`: https://node-postgres.com/api/pool#pool-end

You could also lower the `idleTimeoutMillis` for the pool used for your tests.
